core/storage/suggestion/gae_models.py

```diff
@@ -484,13 +484,43 @@ def get_in_review_translation_suggestions(
             list(SuggestionModel). A list of suggestions that are of the given
             type, which are in review, but not created by the given user.
         """
+
         return cls.get_all().filter(datastore_services.all_of(
             cls.status == STATUS_IN_REVIEW,
             cls.suggestion_type == feconf.SUGGESTION_TYPE_TRANSLATE_CONTENT,
             cls.author_id != user_id,
             cls.language_code.IN(language_codes)
         )).fetch(feconf.DEFAULT_SUGGESTION_QUERY_LIMIT)
 
+    @classmethod
+    def get_in_review_translation_suggestions_with_exp_ids(
+            cls, user_id: str, language_codes: List[str],
+            exp_ids: List[str]
+    ) -> Sequence[GeneralSuggestionModel]:
+        """Gets all translation suggestions which are in review
+        and belong to the passed exp_ids.
+
+        Args:
+            user_id: str. The id of the user trying to make this query.
+                As a user cannot review their own suggestions, suggestions
+                authored by the user will be excluded.
+            language_codes: list(str). The list of language codes.
+            exp_ids: list(str). Exploration IDs matching the target ID of the
+                translation suggestions.
+
+        Returns:
+            list(SuggestionModel). A list of suggestions that are of the given
+            type, which are in review, not created by the given user and
+            specific to the exploration IDs passed.
+        """
+        return cls.get_all().filter(datastore_services.all_of(
+            cls.status == STATUS_IN_REVIEW,
+            cls.suggestion_type == feconf.SUGGESTION_TYPE_TRANSLATE_CONTENT,
+            cls.author_id != user_id,
+            cls.language_code.IN(language_codes),
+            cls.target_id.IN(exp_ids)
```

Comment:
I think App Engine queries _may_ not support this properly (i.e. multiple IN filters). Might need to test it on prod, or restrict to 1 language code (and make separate queries for different language codes).

(IIRC App Engine queries become riskier the more complex filters are added, and sometimes these issues only manifest in prod. One IN should be fine though.)

I'm not sure if it's worth the effort, though. Perhaps make it "one IN" if it's easy, otherwise we should do a [trial run on prod](https://github.com/oppia/oppia/wiki/Testing-jobs-and-other-features-on-production) to make sure we don't hit unexpected failures on deployment.

Reply:
Oh, that's a good point. I will keep this in mind.